src/BinSkim.Rules/EnableSpectreMitigations.cs

```diff
@@ -238,50 +245,73 @@ public override void Analyze(BinaryAnalyzerContext context)
+                    }
 
-                // if either QSpectre or d2guardspecload are enabled AND neither is explicitly disabled then we are protected
-                //      (use of both is confusing so issue an error in this scenario even though they are effectively the same switch)
-                if ((QSpectreState == SwitchState.SwitchEnabled || d2guardspecloadState == SwitchState.SwitchEnabled) && 
-                    (QSpectreState != SwitchState.SwitchDisabled && d2guardspecloadState != SwitchState.SwitchDisabled))
-                {
-                    effectiveState = SwitchState.SwitchEnabled;
-                }
+                    if (QSpectreState == SwitchState.SwitchNotFound && d2guardspecloadState == SwitchState.SwitchNotFound)
+                    {
+                        // built with tools that support the Spectre mitigations but these have not been enabled.
+                        badModuleList.Add(
+                            om.CreateCompilandRecordWithSuffix(
+                                string.Format(CultureInfo.InvariantCulture,
+                                RuleResources.BA2024_Error_BuildWithSpectreMitigation_SpectreMitigationMissing)));
+                        continue;
+                    }
+                    else
+                    {
+                        // built with the Spectre mitigations explicitly disabled.
+                        badModuleList.Add(
+                            om.CreateCompilandRecordWithSuffix(
+                                string.Format(CultureInfo.InvariantCulture,
+                                RuleResources.BA2024_Error_BuildWithSpectreMitigation_SpectreMitigationDisabled)));
+                    }
 
-                if (effectiveState == SwitchState.SwitchDisabled)
-                {
-                    // built with the Spectre mitigations explicitly disabled.
-                    badModuleList.Add(
-                        om.CreateCompilandRecordWithSuffix(
-                            string.Format(CultureInfo.InvariantCulture,
-                            RuleResources.BA2024_Error_BuildWithSpectreMitigation_SpectreMitigationDisabled)));
+                    continue;
                 }
-                else if(effectiveState == SwitchState.SwitchNotFound)
+
+                if (supportsOd == false)
                 {
-                    // built with tools that support the Spectre mitigations but these have not been enabled.
-                    badModuleList.Add(
-                        om.CreateCompilandRecordWithSuffix(
-                            string.Format(CultureInfo.InvariantCulture,
-                            RuleResources.BA2024_Error_BuildWithSpectreMitigation_SpectreMitigationMissing)));
+                    string[] OptimizerSwitches = { "/Od", "/O1", "/O2", "/Ox", "/Og" };
+
+                    bool debugEnabled = false;
```

Comment:
Per the documentation (https://docs.microsoft.com/en-us/cpp/build/reference/od-disable-debug ), /Od is enabled by default--I haven't touched the PDB command line stuff, so I have a clarifying question here:  will the flag show up in the command line parameters in the PDB if it isn't passed at all?

Reply:
No - it will not show up in the command line.

The logic encapsulated by the new functionality in the CompilerCommandLine.GetSwitchStateFactoringOverrides (really not sure if this name conforms with the codebase conventions) allows you to pass a default state which will be set if none of the overriding switches are present.  

So if you are aware of what the default state is you can pass that in, or check for SwitchState.SwitchNotFound

Reply:
Ahh, OK, so the SwitchState.Enabled parameter in the following takes care of that.  Makes sense, thanks!